PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:7 + add:0, change matches:
-    _inherit = 'account.payment'
-    _inherit = 'payment.transaction'
-    _inherit = 'pos.config'
-    _inherit = 'pos.order'
-    _inherit = 'pos.payment'
-    _inherit = "pos.payment.method"
-    _inherit = 'pos.session'

Total Changes: 14

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/pos_online_payment/models/account_payment.py
+++ b/addons/pos_online_payment/models/account_payment.py
@@ -5,7 +5,7 @@ from odoo import _, fields, models
 
 
 class AccountPayment(models.Model):
-    _inherit = 'account.payment'
+    _inherit = ['account.payment']
 
     pos_order_id = fields.Many2one('pos.order', string='POS Order', help='The Point of Sale order linked to this payment', readonly=True)
 

--- a/addons/pos_online_payment/models/payment_transaction.py
+++ b/addons/pos_online_payment/models/payment_transaction.py
@@ -6,7 +6,7 @@ from odoo.exceptions import ValidationError
 
 
 class PaymentTransaction(models.Model):
-    _inherit = 'payment.transaction'
+    _inherit = ['payment.transaction']
 
     pos_order_id = fields.Many2one('pos.order', string='POS Order', help='The Point of Sale order linked to the payment transaction', readonly=True)
 

--- a/addons/pos_online_payment/models/pos_config.py
+++ b/addons/pos_online_payment/models/pos_config.py
@@ -6,7 +6,7 @@ from odoo.exceptions import ValidationError
 
 
 class PosConfig(models.Model):
-    _inherit = 'pos.config'
+    _inherit = ['pos.config']
 
     @api.constrains('payment_method_ids')
     def _check_online_payment_methods(self):

--- a/addons/pos_online_payment/models/pos_order.py
+++ b/addons/pos_online_payment/models/pos_order.py
@@ -5,7 +5,7 @@ from odoo import models, fields, api, tools
 
 
 class PosOrder(models.Model):
-    _inherit = 'pos.order'
+    _inherit = ['pos.order']
 
     online_payment_method_id = fields.Many2one('pos.payment.method', compute="_compute_online_payment_method_id")
     next_online_payment_amount = fields.Float(string='Next online payment amount to pay', digits=0, required=False) # unlimited precision

--- a/addons/pos_online_payment/models/pos_payment.py
+++ b/addons/pos_online_payment/models/pos_payment.py
@@ -9,7 +9,7 @@ _logger = logging.getLogger(__name__)
 
 
 class PosPayment(models.Model):
-    _inherit = 'pos.payment'
+    _inherit = ['pos.payment']
 
     online_account_payment_id = fields.Many2one('account.payment', string='Online accounting payment', readonly=True) # One2one
 

--- a/addons/pos_online_payment/models/pos_payment_method.py
+++ b/addons/pos_online_payment/models/pos_payment_method.py
@@ -6,7 +6,7 @@ from odoo.exceptions import ValidationError
 
 
 class PosPaymentMethod(models.Model):
-    _inherit = "pos.payment.method"
+    _inherit = ["pos.payment.method"]
 
     is_online_payment = fields.Boolean(string="Online Payment", help="Use this payment method for online payments (payments made on a web page with online payment providers)", default=False)
     online_payment_provider_ids = fields.Many2many('payment.provider', string="Allowed Providers", domain="[('is_published', '=', True), ('state', 'in', ['enabled', 'test'])]")

--- a/addons/pos_online_payment/models/pos_session.py
+++ b/addons/pos_online_payment/models/pos_session.py
@@ -7,7 +7,7 @@ from odoo.exceptions import UserError
 
 
 class PosSession(models.Model):
-    _inherit = 'pos.session'
+    _inherit = ['pos.session']
 
     def _accumulate_amounts(self, data):
         data = super()._accumulate_amounts(data)
